src/waypoint.jsx

```diff
@@ -18,14 +18,19 @@ const propTypes = {
   onPositionChange: PropTypes.func,
   fireOnRapidScroll: PropTypes.bool,
   scrollableAncestor: PropTypes.any,
+  // throttle Scroll handler (e.g. 100 ms)
+  throttleHandler: PropTypes.func
 };
 
 const defaultProps = {
   threshold: 0,
   onEnter() {},
   onLeave() {},
   onPositionChange() {},
-  fireOnRapidScroll: true
+  fireOnRapidScroll: true,
+  throttleHandler(handler) {
+    return handler;
+  }
```

Comment:
`throttleHandler` is not that bad, actually we only have one handler, `handleScroll`